### ISSUE TITLE
How do I link a Ghidra Script in Eclipse with the application so that I can debug it?

### ISSUE THREAD

**Issue by itsallmathematics** [NationalSecurityAgency/ghidra#6727]
I have written a **Java** script for Ghidra in Eclipse IDE using GhidraDev GhidraScript. GhidraDev is aware of the Ghidra directory and I am able to access all shipped scripts from within Eclipse. Frankly, everything else seems linked properly. However, when I set a breakpoint and Debug As -> Ghidra, Ghidra launches but the current script file that I am working on is not in Ghidra's Script Manager anywhere. I think this is because the script that I created in Eclipse is in /home/eclipse-workspace/GhidraScripts/bin as .class files, but Ghidra's Script Manager checks $GHIDRA_HOME/Debug/.../ghidra_scripts for .java files. How can I sync these so that the debugger actually will break Ghidra from within the script I'm trying to debug?

**Comment by ryanmkurtz** [comment]
Where does your .java file live on disk? If you didn't make it in '~/ghidra_scripts', you likely need to add your script's directory to the "Bundle Manager" in Ghidra, which is accessible from a button in the Script Manager, as well as a menu item under "Window".

**Comment by itsallmathematics** [comment]
detail: The class could not be found. It must be the public class of the .java file:

> Where does your .java file live on disk? If you didn't make it in `~/ghidra_scripts`, you likely need to add your script's directory to the "Bundle Manager" in Ghidra, which is accessible from a button in the Script Manager, as well as a menu item under "Window".

I guess my confusion here is that in Eclipse, when I right click and click Debug As -> Ghidra, it just launches Ghidra - no JAR file is even made to my knowledge, nor does it place the .java file anywhere special. Do I need to separately export the JAR before debugging? Sorry if this seems obvious, I come from C, Python, C#, and Rust background but not much Java.

**Comment by itsallmathematics** [comment]
> The Bundle Manager wants the directories of where your GhidraScript java source lives. When you run the GhidraScript from Ghidra, Ghidra will compile it behind the scenes before running in.

Thank you. This solution has worked - the issue was that I had another separate issue occurring simultaneous to this issue, and it was obscuring my ability to get this to work properly because I actually did exactly this to begin with, however, there were a bunch of errors, that strangely, providing the classpath directory instead of the .java dir seemed to resolve, but again, that was due to another issue and actually now that that issue is resolved, the dir with the .java files does work as expected.

If anyone is curious, there was an unintentional JDK discrepancy occurring between Ghidra itself and Eclipse, which presumably was causing a bunch of classes and such to not be found and other malfunctions... If anyone else is concerned this may be an issue: Make sure to check the Eclipse Workspace JDK default in Workspace preferences, as well as the classpath/modulepath JDK in Eclipse Package Explorer... Then check the Ghidra splash screen or Help -> About Ghidra to make sure that it's the same JDK as the one in all areas of Eclipse. In my case, there was a strange situation where Ghidra would be launched with the correct matching JDK when not debugging, but when debugging, Eclipse was somehow starting it with another JDK but checking all of these areas, fixing any alias or PATH issues with regard to where `java` was located on the linux work machine, plus finally making sure that the entry in `~/.config/ghidra/ghidra_11.1.2_PUBLIC/java_home.save` matches the proper JDK as well. I don't necessarily recommend changing `java_home.save` directly if it is incorrect though, because the file seems to be opaque and direct changes can be overwritten by ghidra's launching discovery functionality in https://github.com/NationalSecurityAgency/ghidra/tree/master/GhidraBuild/LaunchSupport/src/main/java/ghidra/launch such as JavaFinder, JavaConfig, and so on...

I apprecate your help with this @ryanmkurtz , nice intro to ghidra dev/JDK stuff. :)
